### DISCUSSION TITLE
Follow references when loading documents with CodeFileLoader

### DISCUSSION THREAD

**Discussion by openscript** [dotansimha/graphql-code-generator#7918] [Q&A]
Gatsby is loading some documents via the `CodeFileLoader` [here](https://github.com/gatsbyjs/gatsby/blob/7dfd52d9b43e9fe2e456b273dc75de99634ace5d/packages/gatsby/src/utils/graphql-typegen/ts-codegen.ts#L97). I've been taking `gatsby-node.ts` apart into different files. They all export a function and in the `gatsby-node.ts` I'm awaiting for them:

```typescript
import { GatsbyNode } from 'gatsby';
import { CreateGenericPages } from './src/gatsby/createPages/createGenericPages';

export const createPages: GatsbyNode['createPages'] = async args => {
  await CreateGenericPages(args);
};
```

So the `CoeFileLoader` searches in `gatsby-node.ts` for `graphql` usages, but it doesn't follow the references. Here for example to `CreateGenericPages`. Is there any possibility to make it follow these references or is the approach to let it search the whole `src` anyway?

**Comment by ardatan** [comment]
I am not sure what you mean by "references"? `CodeFileLoader` needs to see an export of GraphQL operation or `gql`/`graphql` tag usage.
Other than that, we cannot expect any more magic from `CodeFileLoader`.
